File: FootballApp/web/football-app-web/src/app/models/column-filter-descriptor.ts
/**
 * One column filter. Mirrors the C# ColumnFilterDescriptor and
 * TanStack Table's ColumnFilter: { id: "country", value: "Spain" }
 *
 * `value` is `unknown` because the API accepts:
 *   - string         (text contains)
 *   - number         (numeric equality)
 *   - [number|null, number|null]  (numeric range)
 *
 * We use `unknown` instead of `any` to force callers to narrow before use.
 */
export interface ColumnFilterDescriptor {
    readonly id: string;
    readonly value: unknown;
}

File: FootballApp/api/FootballApp.Api/Models/ColumnFilterDescriptor.cs
namespace FootballApp.Api.Models;

/// <summary>
/// One column filter. Mirrors TanStack Table's ColumnFilter:
///   { id: "country", value: "Spain" }
///
/// Value is intentionally typed as object? because TanStack lets the
/// filter value be any JSON-serializable thing (string, number, bool,
/// array, range object). The service layer interprets it per column.
/// </summary>
public sealed class ColumnFilterDescriptor
{
    public string Id { get; set; } = string.Empty;

    public object? Value { get; set; }
}